### PR TITLE
Aws documentation update

### PR DESCRIPTION
**Pull request by ballastlane-ivanramirez**

### CODE REVIEW

File: src/documentation.js [hunk 1]
```diff
@@ -26,13 +26,70 @@ function _mapToObj(map) {
   return returnObj;
 }
 
+/**
+ * maps requests to delete to an array containing the promise and the body of the request
+ * @param {aws} provider
+ * @param {documentationPieces} Object, contains documentationToUpsert and documentationToDelete
```
Comment (martin-dessureault) [inline src/documentation.js:32]: please update: remove documentationPieces and add documentationToUpsert and documentationToDelete

File: src/documentation.js [hunk 2]
```diff
@@ -26,13 +26,70 @@ function _mapToObj(map) {
...
+async function resolveDocumentationRequests(requests) {
+  console.log("---- resolveDocumentationRequests ----");
+  let duplicates = 0
+  for (let index = 0; index < requests.length; index++) {
+    const request = requests[index];
+    request.request
+      .then(data => console.log(data))
+      .catch((error) => {
+        if (error.providerErrorCodeExtension === "CONFLICT_EXCEPTION") {
+          duplicates = duplicates + 1
```
Comment (martin-dessureault) [inline src/documentation.js:77]: `duplicates += 1`

File: src/documentation.js [hunk 3]
```diff
@@ -26,13 +26,70 @@ function _mapToObj(map) {
...
+        }),
+        part
+      }
+    })
+  )
+  // upsert documentation parts
+  requests = requests.concat(
+    documentationToUpsert.map(part => {
+      return {
+        request: aws.request('APIGateway', 'createDocumentationPart', {
```
Comment (martin-dessureault) [inline src/documentation.js:53]: I believe this is where we are getting rate limited. anyway to catch those and slow down?

File: src/documentation.js (hunk 1 above)
Comment (ballastlane-ivanramirez) [reply src/documentation.js:32]: read a couple things on JSDoc, I'll update comments

File: src/documentation.js [hunk 4]
```diff
@@ -26,13 +26,71 @@ function _mapToObj(map) {
...
+ * @param {aws} provider
+ * @param {documentationPieces} Object, contains documentationToUpsert and documentationToDelete
+ * @returns request type array
+ */
+function createDocumentationRequests(aws, { documentationToUpsert, documentationToDelete }) {
+  let requests = []
+  // delete documentation parts
+  requests = requests.concat(
+    documentationToDelete.map(part => {
+      return{
```
Comment (martin-dessureault) [inline src/documentation.js]: nit: return` ` {

File: src/documentation.js [hunk 5]
```diff
@@ -164,45 +208,48 @@ module.exports = function() {
...
+        documentationToDelete
+      })
+      
+      try {
+        console.log("sending documentation requests");
+        console.log("documentation request concurrency: ", concurrency);
+        let i = 0
+        while (requests.length) {
+          i += 1
+          console.log("iterations: ", i);
```
Comment (martin-dessureault) [inline src/documentation.js]: nit: `console.log("iterations: ", ++i);`

File: src/documentation.js [hunk 6]
```diff
@@ -125,8 +168,9 @@ module.exports = function() {
         return loc;
       }, {});
       location.type = part.type;
+      
       const propertiesToGet = determinePropertiesToGet(location.type)
-
+
```
Comment (martin-dessureault) [inline src/documentation.js]: nit: remove extra spaces

**Review by martin-dessureault — DISMISSED** [review]
minor suggestions that doesn't block merging

**Review by martin-dessureault — APPROVED** [review]

**Pull request merged**